README.md

```diff
@@ -7,131 +7,330 @@
+* If value of masked key corresponds to an `object`, all nested fields, including nested arrays and objects) will be masked recursively
+* If value of masked key corresponds to an `array`, all values of the array (including nested arrays and objects) will be masked recursively
+* Ability to define different masking strategy per type
+   - **(default)** mask strings with a different string: `"maskMe": "secret"` -> `"maskMe": "***"`
+   - mask _characters_ of a string with a different character: `"maskMe": "secret"` -> `"maskMe": "*****"` (preserves length)
+   - **(default)** mask numbers with a string: `"maskMe": 12345` -> `"maskMe": "###"` (changes number type to string)
+   - mask numbers with a different number: `"maskMe": 12345` -> `"maskMe": 0` (preserves number type)
+   - mask _digits_ of a number with a different digit: `"maskMe": 12345` -> `"maskMe": 88888` (preserves number type and length)
+   - **(default)** mask booleans with a string: `"maskMe": true` -> `"maskMe": "&&&"` (changes boolean type to string)
+   - mask booleans with a different boolean: `"maskMe": true` -> `"maskMe": false` (preserves boolean type)
+* Ability to define masking strategy per key
 * Target key **case sensitivity configuration** (default: `false`)
-* **Block-list** (`masked`) or **allow-list** (`allowed`) interpretation of target key set (default: `masked`)
-* Masking valid JSON will always result in valid JSON
-* The implementation only supports JSON in UTF-8 character encoding
+* Use **block-list** (`maskKeys`) or **allow-list** (`allowKeys`) for masking
+* Limited support for JsonPath masking in **block-list** (`maskJsonPaths`) and **allow-list** (`allowJsonPaths`) modes
+* Masking a valid json will always return a valid json
+* The implementation only supports json in UTF-8 character encoding
 
 ## Usage examples
 
+`JsonMasker` instance can be created using of of the following factory methods:
+```java
+// block-mode, default config
+var jsonMasker = JsonMasker.getMasker(Set.of("email", "iban"));
+
+// block-mode, default config (using a builder)
+var jsonMasker = JsonMasker.getMasker(
+        JsonMaskingConfig.builder()
+                .maskKeys(Set.of("email", "iban"))
+                .build()
+);
+
+// block-mode, json path
+var jsonMasker = JsonMasker.getMasker(
+        JsonMaskingConfig.builder()
+                .maskJsonPaths(Set.of("$.email", "$.nested.iban"))
+                .build()
+);
+
+// allow-mode, default config
+var jsonMasker = JsonMasker.getMasker(
+        JsonMaskingConfig.builder()
+                .allowKeys(Set.of("id", "name"))
+                .build()
+);
+
+// allow-mode, json path
+var jsonMasker = JsonMasker.getMasker(
+        JsonMaskingConfig.builder()
+                .maskJsonPaths(Set.of("$.id", "$.nested.name"))
+                .build()
+);
+```
+
+Using `JsonMaskingConfig` allows customizing the masking behaviour of types, keys or json path or mix keys and json paths.
+
+> [!NOTE]
```

Comment:
Would also make sense to specify that jsonpath masking config takes precedent before simple key masking config (in a situation when jsonpath key and simple key point to the same key/value in the json)